### PR TITLE
treewide: replace make/build/configure/patchFlags with nix lists

### PR DESCRIPTION
**Pull request by merijnb**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @globin Contribution from Nixcon Hackathon.

This commit wasn't even compiled, so will likely have syntax errors. It was made by running a crude sequence of sed steps, after discussing with @globin at Nixcon. These were the sed steps for example for one set of flags:
```
rg -l 'configureFlags = "' | xargs -n 1 sed -i 's/Flags = "\([^ ]*\) \(.*\)"/Flags = "\1" "\2"/'
rg -l 'configureFlags = "' | xargs -n 1 sed -i 's/Flags = "\([^ ]*\)" "\([^ ]*\) \(.*\)"/Flags = "\1" "\2" "\3"/' 
rg -l 'configureFlags = "' | xargs -n 1 sed -i 's/Flags = "\([^ ]*\)" "\([^ ]*\)" "\([^ ]*\) \(.*\)"/Flags = "\1" "\2" "\3" "\4"/' 
rg -l 'configureFlags = "[^ ]*"' | xargs -n 1 sed -i 's/Flags = "\(.*\)"/Flags = \[ "\1" \]/'
```

This gets quite a few constructs wrong, I've fixed the ones I've noticed wrong. It will get legitimate spaces inside the strings wrong, or if there is an if else on the same line. 

I'll make a small python program that will work better, so other flags can be changed by @globin when he comes across more.

### CODE REVIEW

**Comment by jonringer** [comment]
is that the right base branch?

File: doc/languages-frameworks/go.xml [hunk 1]
```diff
@@ -94,7 +94,7 @@ deis = buildGoPackage rec {
 
   goDeps = ./deps.nix; <co xml:id='ex-buildGoPackage-3' />
 
-  buildFlags = "--tags release"; <co xml:id='ex-buildGoPackage-4' />
+  buildFlags = [ "--tags" "release" ]; <co xml:id='ex-buildGoPackage-4' />
```
Comment (FRidh) [inline doc/languages-frameworks/go.xml]: not ok

**Review by FRidh — CHANGES_REQUESTED** [review]

File: pkgs/applications/audio/aeolus/default.nix [hunk 2]
```diff
@@ -20,7 +20,7 @@ stdenv.mkDerivation rec {
 
   preBuild = "cd source";
 
-  makeFlags = "DESTDIR= PREFIX=$(out)";
+  makeFlags = [ "DESTDIR=" "PREFIX=$(out)" ];
```
Comment (FRidh) [inline pkgs/applications/audio/aeolus/default.nix]: Should we use placeholder?

**Comment by merijnb** [comment]
> is that the right base branch?

This is debatable. It is intentional, to help the structured-attrs branch to move forward and not be hung up on packages breaking for these small issues. Leaving this in that branch (which is not intended to hit until after 20.03) will make for a lot of churn to keep it in sync with master, so cherry picking this forward earlier seems like a good idea.

**Comment by FRidh** [comment]
If I understand correctly it is a dependency for structured attributes. It is however a good change to get in anyway and is thus best merged as soon as possible.

File: doc/languages-frameworks/go.xml (hunk 1 above)
Comment (merijnb) [reply doc/languages-frameworks/go.xml]: Can you expand on what this issue is here?

Comment (FRidh) [reply doc/languages-frameworks/go.xml]: oh, sorry, it is fine, I misread it.

File: pkgs/applications/audio/aeolus/default.nix (hunk 2 above)
Comment (globin) [reply pkgs/applications/audio/aeolus/default.nix]: fine for now

**Comment by globin** [comment]
yeah this is just help] for my work on #72074, thanks @merijnb!

**Pull request merged**